### PR TITLE
improve solpos tz requirements

### PR DESCRIPTION
**Pull request by wholmgren**
- [ ] Closes #xxxx
 - [ ] I am familiar with the [contributing guidelines](https://pvlib-python.readthedocs.io/en/latest/contributing.html)
 - [ ] Tests added
 - [ ] Updates entries to [`docs/sphinx/source/api.rst`](https://github.com/pvlib/pvlib-python/blob/master/docs/sphinx/source/api.rst) for API changes.
 - [ ] Adds description and name entries in the appropriate "what's new" file in [`docs/sphinx/source/whatsnew`](https://github.com/pvlib/pvlib-python/tree/master/docs/sphinx/source/whatsnew) for all changes. Includes link to the GitHub Issue with `` :issue:`num` `` or this Pull Request with `` :pull:`num` ``. Includes contributor name and/or GitHub username (link with `` :ghuser:`user` ``).
 - [ ] New code is fully documented. Includes [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) compliant docstrings, examples, and comments where necessary.
 - [x] Pull request is nearly complete and ready for detailed review.
 - [ ] Maintainer: Appropriate GitHub Labels and Milestone are assigned to the Pull Request and linked Issue.

We've seen several issues recently that are due to users misunderstanding the requirements for timezone specification in solar position calculations. Here's [one example](https://groups.google.com/d/msg/pvlib-python/aa-bZPXuohc/V5NiAOvyDgAJ)

This is a quick first step at improving the documentation by filling in missing parts of the docs strings. 

@mikofski @cwhanse thoughts on this? I'm happy to make small changes but would like to keep the scope of this PR fairly limited.

### CODE REVIEW

**Comment by wholmgren** [comment]
Ok to merge @cwhanse?

File: pvlib/solarposition.py [hunk 1]
```diff
@@ -137,13 +144,17 @@ def spa_c(time, latitude, longitude, pressure=101325, altitude=0,
...
+        Latitude in decimal degrees. Positive north of equator, negative
+        to south.
     longitude : float
+        Longitude in decimal degrees. Positive east of prime meridian,
+        negative to west.
     pressure : float, default 101325
         Pressure in Pascals
     altitude : float, default 0
-        Elevation above sea level.
+        Distance above sea level.
```
Comment (cwhanse) [inline pvlib/solarposition.py]:
```suggestion
        Height above sea level.
```

File: pvlib/solarposition.py [hunk 2]
```diff
@@ -574,11 +590,13 @@ def pyephem(time, latitude, longitude, altitude=0, pressure=101325,
...
     latitude : float
-        positive is north of 0
+        Latitude in decimal degrees. Positive north of equator, negative
+        to south.
     longitude : float
-        positive is east of 0
+        Longitude in decimal degrees. Positive east of prime meridian,
+        negative to west.
     altitude : float, default 0
         distance above sea level in meters.
```
Comment (cwhanse) [inline pvlib/solarposition.py]:
```suggestion
        Height above sea level in meters. [m]
```

**Comment by wholmgren** [comment]
Thanks for the review. I'll go ahead and merge.

**Comment by mikofski** [comment]
thanks Will!

**Pull request merged**